### DISCUSSION TITLE
Start hotspot => No Internet

### DISCUSSION THREAD

**Discussion by hsmptg** [RaspAP/raspap-webgui#931] [Q&A]
Hi
I found several posts about this problem and believe me I tried all the tips but with no luck!

I have a Raspberry Pi 2 Model B v1.1 with two external WiFi USB dongles.
It has the latest 2021-05-07-raspios-buster-armhf-lite image fully updated.
Before installing RaspAP I can ssh into the RPi using the 3 interfaces (eth0, wlan0 and wlan1), and from there I can access the Internet using a ping 1.1.1.1.
After installing the RaspAP v2.6.6 using the Quick installer, and starting the hotspot in the wlan0 I obtain the desire AP but I loose the access to the internet if I ssh using the wlan1 (the internet keeps working if I ssh using the eth0). If I stop the hotspot the internet immediately becomes OK again in the wlan1.

In my application I can not use the eth0 and therefore I need to have Internet access through wlan1 when the wlan0's hotspot is on. Can you please help me on this?

As I said, some of the tips I saw are related with the iptables, something that I also tried but that I think that are already implemented in the current Quick installer, right?

Regards,
Hélio (hsmptg)

**Comment by hsmptg** [comment]
Hi
This is the result of a ping 1.1.1.1 along several hotspot actions.
I initiate the ping with the hotspot started and therefore I do not have internet access:

pi@rpi-falesia:~ $ ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
From 10.3.141.1 icmp_seq=1 Destination Host Unreachable
From 10.3.141.1 icmp_seq=2 Destination Host Unreachable
From 10.3.141.1 icmp_seq=3 Destination Host Unreachable
From 10.3.141.1 icmp_seq=4 Destination Host Unreachable
From 10.3.141.1 icmp_seq=5 Destination Host Unreachable
From 10.3.141.1 icmp_seq=6 Destination Host Unreachable
64 bytes from 1.1.1.1: icmp_seq=10 ttl=57 time=7.94 ms <- here I stop the hotspot
64 bytes from 1.1.1.1: icmp_seq=11 ttl=57 time=8.16 ms
64 bytes from 1.1.1.1: icmp_seq=12 ttl=57 time=7.88 ms
64 bytes from 1.1.1.1: icmp_seq=13 ttl=57 time=7.84 ms
64 bytes from 1.1.1.1: icmp_seq=14 ttl=57 time=7.51 ms
64 bytes from 1.1.1.1: icmp_seq=15 ttl=57 time=7.66 ms
64 bytes from 1.1.1.1: icmp_seq=16 ttl=57 time=8.81 ms
64 bytes from 1.1.1.1: icmp_seq=17 ttl=57 time=7.76 ms <- here I start the hotspot (Executing RaspAP service start...)
64 bytes from 1.1.1.1: icmp_seq=18 ttl=57 time=7.90 ms
ping: sendmsg: Network is unreachable
ping: sendmsg: Network is unreachable
ping: sendmsg: Network is unreachable
ping: sendmsg: Network is unreachable
ping: sendmsg: Network is unreachable
ping: sendmsg: Network is unreachable
ping: sendmsg: Network is unreachable
64 bytes from 1.1.1.1: icmp_seq=26 ttl=57 time=11.2 ms
64 bytes from 1.1.1.1: icmp_seq=27 ttl=57 time=7.67 ms
64 bytes from 1.1.1.1: icmp_seq=28 ttl=57 time=7.66 ms
From 10.3.141.1 icmp_seq=29 Destination Host Unreachable
From 10.3.141.1 icmp_seq=30 Destination Host Unreachable
From 10.3.141.1 icmp_seq=31 Destination Host Unreachable
From 10.3.141.1 icmp_seq=32 Destination Host Unreachable
From 10.3.141.1 icmp_seq=33 Destination Host Unreachable
From 10.3.141.1 icmp_seq=34 Destination Host Unreachable

As you can see when I start the hotspot and the RaspAP start the "Executing RaspAP service start..." procedure, I still have internet access for brief seconds, then I loose connection, then I get it back, and when the "Executing RaspAP service start..." procedure closes I lost the access definitively (it only returns if I stop the hotspot again).

Any clue on what is causing this?

Regards
Hélio (hsmptg)

**Comment by hsmptg** [comment]
Meanwhile I tried to use a Raspberry Pi 4, this time with a single external USB WiFi dongle since the Raspberry has an internal WiFi interface, but unfortunately the problem remains: when I start the hotspot the internet access is lost!!! :(

I know that probably the majority of cases using RaspAP use the eth0 to access the internet (not a WiFi interface) but I bet I am not the only one with this setup (no etho + 2 wlan) so... anyone can help me please?

Regards
Hélio (hsmptg)

**Comment by billz** [comment]
Relevant FAQ link for starters https://docs.raspap.com/faq/#interfaces
Original step-by-step instructions https://github.com/RaspAP/raspap-webgui/issues/580#issuecomment-642970534

Is IP forwarding enabled on your Pi? 
```
cat /proc/sys/net/ipv4/ip_forward
```
should return a value of 1.

RaspAP has built-in support for wlan0/wlan1 "wireless repeater" mode via the UI, ie. no manual configuration or other changes required. If this is not a clean install, consider starting with a fresh SD card.

**Reply by hsmptg** [reply]
Hi
First of all thank you for the reply!

> Relevant FAQ link for starters https://docs.raspap.com/faq/#interfaces
> Original step-by-step instructions [#580 (comment)](https://github.com/RaspAP/raspap-webgui/issues/580#issuecomment-642970534)
> 
> Is IP forwarding enabled on your Pi?
> 
> ```
> cat /proc/sys/net/ipv4/ip_forward
> ```
> 
> should return a value of 1.

Yes, it returns a 1

> RaspAP has built-in support for wlan0/wlan1 "wireless repeater" mode via the UI, ie. no manual configuration or other changes required. If this is not a clean install, consider starting with a fresh SD card.

If I accept all the default settings and just start the hotspot, I have wlan1 as the WiFi CLient and wlan0 as the hotspot. The problem is that with the hotspot started I can not ping inside a ssh using wlan1.

If I see the rules.v4 I get this:

```
pi@rpi-four:~ $ sudo cat /etc/iptables/rules.v4
# Generated by xtables-save v1.8.2 on Sun Jun 13 22:56:53 2021
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
COMMIT
# Completed on Sun Jun 13 22:56:53 2021
# Generated by xtables-save v1.8.2 on Sun Jun 13 22:56:53 2021
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
-A POSTROUTING -j MASQUERADE
-A POSTROUTING -s 192.168.50.0/24 ! -d 192.168.50.0/24 -j MASQUERADE
COMMIT
# Completed on Sun Jun 13 22:56:53 2021
```

I do not see any mentions to wlan0 or wlan1 so I am not surprise for wlan1 can not access internet.

Instead of using RaspAP I tried to follow this procedure (https://www.raspberrypi.org/documentation/configuration/wireless/access-point-routed.md) and if I replace eth0 for wlan1 I was able to have internet access but it uses the following rules:

```
sudo iptables -t nat -A POSTROUTING -o wlan1 -j MASQUERADE
```

and two other that I found in a couple of posts:

```
sudo iptables -A FORWARD -i wlan1 -o wlan0 -m state --state RELATED,ESTABLISHED -j ACCEPT
sudo iptables -A FORWARD -i wlan0 -o wlan1 -j ACCEPT
```

So at least this way I already have what I needed!

Regards
Hélio (hsmptg)

**Reply by yoshideryu** [reply]
i had this same issue and made a custom menu called routes and added buttons for each interface and OpenVPN tunnel. It also displays the interface info the iptables and route tables in my case i would loss internet when connecting to VPN.  i also added a on boot cron job to start with my internet access this has made me happy and no longer need RaspAP to control my route tables and iptables 

my setup is basically wlan1 is the hotspot and wlan0 is the internet connected to whatever ssid is in the area '

i use my pi as a vpn router for my tvs  when i travel to japan many things are locked and i need vpn to view

**Reply by vplikin** [reply]
Thanks, worked for me too. Totally forgot about iptables and masquerading at all
This tweak should be in the docs already, but I didnt find it
